docs/user_guide/modal-publishing.md

```diff
@@ -0,0 +1,151 @@
+## 1. Create a Modal Account
+
+**NOTE:** This step requires a free [GitHub](https://github.com) account to authenticate with Modal.
+
+Create a free Modal account: [signup](https://modal.com/signup)
+
+## 2. Author a Modal App
+
+### Create a Modal file
+
+Modal files are regular python scripts. Create a new python file in your project:
+
+```bash
+$ touch my_app.py
+```
+
+### Imports
+
+```python
+# my_app.py
+
+# Import the Modal SDK
+import modal
+```
+
+### Define the Modal App
+
+Create a `modal.App` obeject where your custom functions will be registered.
+Modal apps need to be assigned to a variable named `app` and in the global scope.
+
+```
+app = modal.App("my-cool-app")
+```
+
+### Define Modal functions
+
+Modal functions are regular python functions that have been decorated with `@app.function()`.
+The `@app.function()` decorator registers the function with the `modal.App` created above.
+Like Modal Apps, Modal functions need to be defined in the global scope.
+
+```python
+# Define a function and register it with the app
+# Functions can be named anything you like
+@app.function()
+def my_awesome_function(data):
+  result = sum(data)
+  return result
+
+
+# You can register multiple functions to the same app
+@app.function()
+def my_other_cool_function(data):
+  # you can call modal functions from within other modal functions
+  inital_results = my_awesome_function.remote(data)
```

Comment:
I think we should cut this example, we don't want our users trying to call modal functions from inside other modal functions since that would make it easy to dodge usage limits

Reply:
Good catch! I didn't think of that!